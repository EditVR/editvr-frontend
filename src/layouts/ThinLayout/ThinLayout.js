/**
 * @file ThinLayout.js
 * Exports a React component that renders EditVR's thin layout.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core';

import { ThemeProvider } from '../../hoc';
import { Header, Footer } from '../';
import ThinLayoutStyles from './ThinLayout.style';

const ThinLayout = ({ children, classes }) => (
  <div id="layout__wrapper" className={classes.wrapper}>
    <div id="layout__thin-wrapper" className={classes.thinWrapper}>
      <Header />
      {children}
      <Footer />
    </div>
  </div>
);

ThinLayout.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.shape({
    wrapper: PropTypes.string.isRequired,
    thinWrapper: PropTypes.string.isRequired
  }).isRequired
};

ThinLayout.defaultProps = {
  children: null
};

export default ThemeProvider(withStyles(ThinLayoutStyles)(ThinLayout));
