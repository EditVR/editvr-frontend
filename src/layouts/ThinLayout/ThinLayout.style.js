/**
 * @file Layout.style.js
 * Exports styling that's shared across many layouts.
 */

export default theme => ({
  wrapper: {
    background: theme.palette.grays['500'],
    width: '100%',
    height: '100%',
    position: 'absolute',
    overflowX: 'scroll'
  },
  thinWrapper: {
    padding: '0 1rem',
    maxWidth: 600,
    margin: '0 auto',
    paddingTop: '10%'
  }
});
