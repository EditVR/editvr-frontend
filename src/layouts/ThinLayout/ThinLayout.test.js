/**
 * @file ThinLayout.test.js
 * Contains tests for ThinLayout.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import ThinLayout from './ThinLayout';

describe('<ThinLayout />', () => {
  it('Matches its snapshot', () => {
    expect(
      renderer
        .create(
          <ThinLayout>
            <div>child</div>
          </ThinLayout>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
