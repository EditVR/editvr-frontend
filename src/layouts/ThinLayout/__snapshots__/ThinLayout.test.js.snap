// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThinLayout /> Matches its snapshot 1`] = `
<div
  className="ThinLayout-wrapper-1"
  id="layout__wrapper"
>
  <div />
  <div
    className="ThinLayout-thinWrapper-2"
    id="layout__thin-wrapper"
  >
    <header
      className="Header-wrapper-3"
    >
      <img
        alt="EditVR logo"
        className="Header-logo-4"
        src="editvr-logo.svg"
      />
    </header>
    <h1
      className="MuiTypography-root-5 MuiTypography-headline-10"
    >
      test title
    </h1>
    <div>
      child
    </div>
    <footer
      className="Footer-wrapper-29"
    >
      <div
        className="Footer-elements-30"
      >
        <a
          href="https://fourkitchens.com"
        >
          <p
            className="MuiTypography-root-5 MuiTypography-body1-14 Footer-text-32"
            type="body1"
          >
            EditVR is a product of
          </p>
          <img
            alt="Four Kitchens"
            className="Footer-logo-31"
            src="4k-logo-reversed.svg"
          />
        </a>
      </div>
    </footer>
  </div>
</div>
`;
